### PR TITLE
OneBranch Builds Legacy-SDK

### PR DESCRIPTION
**Pull request by littleaj**
* Adds OneBranch .pipelines folder with build definitions
* Adds .scripts folder with build helper scripts (installs dependencies, and other env setup, if needed in the future).
* Adds dependency lockfiles because dependency scan is done in parallel w/o an opportunity to generate on the fly.
* collectd plugin and other artifacts are separated. collectd-api.jar is built on demand and requires linux. other artifacts require windows.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -80,6 +80,25 @@ allprojects {
         }
     }
 
+    tasks.register('resolveDependencies') {
```
Comment (littleaj) [inline build.gradle:83]: this is so we can resolve dependencies during the restore stage

File: buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy [hunk 2]
```diff
@@ -32,9 +32,10 @@ class PropsFileGen extends DefaultTask {
     private def generate() {
         def targetDir = targetFile.getParentFile()
         targetDir.mkdirs()
-        FileOutputStream out = new FileOutputStream(targetFile.getAbsolutePath())
-        props.store(out, comment)
-        out.close()
+        def p = props
```
Comment (littleaj) [inline buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy:35]: do this in a more groovy way, see below

File: buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy [hunk 3]
```diff
@@ -43,8 +44,9 @@ class PropsFileGen extends DefaultTask {
         }
 
         Properties oldProps = new Properties();
-        FileInputStream inFileStream = new FileInputStream(targetFile);
-        oldProps.load(inFileStream);
+        targetFile.withInputStream { inFileStream ->
```
Comment (littleaj) [inline buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy:47]: do this in a groovy way, and also, close the stream so the build doesn't hang :)

**Comment by heyams** [comment]
@trask E2E works now.  Please review.  Need help with making buggy-legacy-sdk as required.. right now settings/branches doesn't load it somehow.

File: .pipelines/pipeline.user.windows.buddy.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      arguments: ':core:windowsSharedLibrary'
+      artifacts:
+        - from: 'core/build/libs/windows/shared'
+          include:
+            - '**/*'
+          exclude:
+            - '**/*.pdb'
+          signing_options:
+            profile: external_distribution
+            sign_inline: true
```
Comment (trask) [inline .pipelines/pipeline.user.windows.buddy.yml:21]: can the buddy builds sign things?

File: .pipelines/pipeline.user.windows.official.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            - '*WindowsCpp/*'
+    - !!buildcommand
+      name: 'Assemble ApplicationInsights-Java JARs'
+      command: '.scripts/gradle.cmd'
+      arguments: 'assemble prepare -Pcore.native.artifacts.prebuilt=true -DisRelease=true'
+      artifacts:
+        - to: 'Artifacts'
+          include:
+            - '**/build/libs/*.jar'
+            - '**/build/libs/*.jar'
```
Comment (trask) [inline .pipelines/pipeline.user.windows.official.yml:38]: duplicate line
```suggestion
```

File: .pipelines/pipeline.user.windows.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        - title: ':logging:log4j2 Unit Test Results'
+          type: 'junit'
+          from: 'logging/log4j2/build/test-results/test'
+          include:
+            - '**/*'
+        - title: ':logging:logback Unit Test Results'
+          type: 'junit'
+          from: 'logging/logback/build/test-results/test'
+          include:
+            - '**/*'
```
Comment (trask) [inline .pipelines/pipeline.user.windows.yml:111]: can we do same as 3.0 build?
```suggestion
        - title: ':Unit Test Results'
          type: 'junit'
          include:
            - '**/build/test-results/test/**/*'
```

File: .pipelines/pipeline.user.windows.buddy.yml (hunk 4 above)
Comment (heyams) [reply .pipelines/pipeline.user.windows.buddy.yml:21]: it was signing 3rd party dll, it used external_distribution.  it works on the buddy build though.. i didn't find any errors so far.

Comment (heyams) [reply .pipelines/pipeline.user.windows.buddy.yml:21]: i can remove this.. signing buddy build doesn't offer much benefits.

Comment (heyams) [reply .pipelines/pipeline.user.windows.buddy.yml]: it will save some build time :)

**Review by trask — APPROVED** [review]

File: test/smoke/testApps/build.gradle [hunk 7]
```diff
@@ -121,5 +121,8 @@ subprojects {
 tasks.register("listTestApps") {
     group = "Help"
     description = "List all available test apps"
-    subprojects.collect { it.path }.each { println it }
+    doFirst {
```
Comment (heyams) [inline test/smoke/testApps/build.gradle]: @trask doFirst works now.. curious why?

**Comment by heyams** [comment]
I think it was a sporadic failure on verifyJava7.  Rerun jobs.

File: test/smoke/testApps/build.gradle (hunk 7 above)
Comment (trask) [reply test/smoke/testApps/build.gradle]: I don't remember, I was able to change this back in the other PR too

File: test/smoke/testApps/VerifyJava7/src/main/java/com/microsoft/ajl/simple/TestController.java [hunk 8]
```diff
@@ -40,7 +40,7 @@ public String verifyJava7() throws IOException {
             if (name.endsWith(".class")) {
                 VersionCapturingClassVisitor cv = new VersionCapturingClassVisitor();
                 new ClassReader(jarIn).accept(cv, 0);
-                if (name.equals("com/microsoft/applicationinsights/core/dependencies/xmlpull/mxp1/MXParser.class") && cv.version == 196653) {
+                if (name.equals("com/microsoft/applicationinsights/core/dependencies/xmlpull") && cv.version == 196653) {
```
Comment (trask) [inline test/smoke/testApps/VerifyJava7/src/main/java/com/microsoft/ajl/simple/TestController.java:43]:
```suggestion
                if (name.startsWith("com/microsoft/applicationinsights/core/dependencies/xmlpull/") && cv.version == 196653) {
```

**Pull request merged**